application/client/src/app/ui/views/toolbar/search/input/input.ts

```diff
@@ -95,6 +96,27 @@ export class SearchInput {
         }
     }
 
+    public onFlag(): {
+        caseSensitive(): void;
+        wholeWord(): void;
+        regex(): void;
+    } {
+        return {
+            caseSensitive: () => {
+                this.flags.cases = !this.flags.cases;
+                this.actions.accept.emit();
+            },
+            wholeWord: () => {
+                this.flags.word = !this.flags.word;
+                this.actions.accept.emit();
+            },
+            regex: () => {
+                this.flags.reg = !this.flags.reg;
+                this.actions.accept.emit();
+            },
+        };
+    }
+
```

Comment:
The general issue is: you are not tracking a state of active search. If you do not create a search (leave input empty) and switch flags - an empty search would be triggered. Reproducing:

1. Open any file
2. Do not write any in search-input, but just click on any flag-icon

Result: empty search is triggered.

Also from a code point of view, it still looks not accomplished. We have state of input `application/client/src/app/ui/views/toolbar/search/input/input.ts` and we have active search `application/client/src/app/ui/views/toolbar/search/input/active.ts`. I think if active-search was changed, we should change also `ActiveSearch`, probably with some kind of `update()` method, maybe. 

Probably solution could be addition event in `application/client/src/app/ui/views/toolbar/search/input/input.ts` - `flags`.

```
export class SearchInput {
    ...
    public flags: IFilterFlags = {
        word: false,
        cases: false,
        reg: true,
    };
    public actions: {
        ...
    } = {
        drop: new Subject(),
        clear: new Subject(),
        accept: new Subject(),
        recent: new Subject(),
        edit: new Subject(),
        flags: new Subject(),
    };

    ...

    // (!) This method can be refactored to have all setters in one place
    
    // public set(value: string | IFilter) {
    //     if (typeof value === 'string') {
    //         this.control.setValue(value);
    //         this._prev = value;
    //     } else {
    //         this.control.setValue(value.filter);
    //         this.flags = obj.clone(value.flags);
    //     }
    // }

    ...

    public set(): {
        value(value: string | IFilter): void;
        caseSensitive(): void;
        wholeWord(): void;
        regex(): void;
    } {
        return {
            value: (value: string | IFilter): void => {
                if (typeof value === 'string') {
                    this.control.setValue(value);
                    this._prev = value;
                } else {
                    this.control.setValue(value.filter);
                    this.flags = obj.clone(value.flags);
                }
            },
            caseSensitive: () => {
                this.flags.cases = !this.flags.cases;
                this.actions.flags.emit();
            },
            wholeWord: () => {
                this.flags.word = !this.flags.word;
                this.actions.flags.emit();
            },
            regex: () => {
                this.flags.reg = !this.flags.reg;
                this.actions.flags.emit();
            },
        };
    }
}
```
In this case in `component` you can safely listen it:
```
    public ngAfterViewInit(): void {
        ...
        this.input.actions.accept.subscribe(() => {
            const filter = this.input.asFilter();
            this.recent.update(filter.filter);
            this.session.search
                .state()
                .setActive(filter)
                .then(() => {
                    this.active = new ActiveSearch(filter);
                    this.input.drop();
                })
                .finally(() => {
                    this.markChangesForCheck();
                });
        });
        this.input.actions.flags.subscribe(() => {
            if (this.active === undefined) {
                return;
            }
            this.active.setFlags(this.input.flags);
            this.session.search
                .state()
                .setActive(this.active.filter)
                .finally(() => {
                    this.markChangesForCheck();
                });
        });
        ...
    }
```

Please, take into account (!) That's only suggestion of a solution, you can do another solution for sure.

Note (!). The code in my comment cannot be used as code of solution: it isn't checked, isn't tested and doesn't consider other goals of this PR. It's just an illustration of a possible solution.

Reply:
While you are here, you can probably also take care restoring of search-progress. Situation is next: user started "heavy" search - we show progress; user left tab and back again - no search progress even it's still in progress. I guess we need a progress flag in search state.